src/etc/inc/interfaces.inc

```diff
@@ -6000,12 +6005,13 @@ function get_interface_vendor_mac($interface) {
 	global $g;
 
 	$mac = "";
-	$mac_file = "{$g['cf_conf_path']}/vendor_mac_{$interface}";
+	$mac_file = "{$g['vardb_path']}/vendor_mac";
 	if (file_exists($mac_file)) {
-		$mac = file_get_contents($mac_file);
-		if (!is_macaddr($mac)) {
-			$mac = "";
-		}
+		$vendor_mac_contents = file_get_contents($mac_file);
+		$mac_pattern = '(?:[[:xdigit:]]{2}:){5}[[:xdigit:]]{2}';
+		$pattern = '/^'.$interface.': ('.$mac_pattern.')$/m';
```

Comment:
Oh, I just thought a second after I submitted previous comment. What about use json_encode() and json_decode(), this way PHP can convert it to an array even easier